Platform/ElkhartlakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c

```diff
@@ -812,6 +814,13 @@ BoardInit (
     }
     break;
   case ReadyToBoot:
+    if (FeaturePcdGet (PcdFspNoEop)) {
+      MeEndOfPostEvent (&NeedReboot);
```

Comment:
For OSloader boot, this should be invoked from OSloader.
UEFI payload also should be done similar to Bios, register a callback to  EfiCreateEventReadyToBootEx.
Incase of universal payload and others, it can be done in SBL.

Reply:
Let me verify if UEFI Payload goes current path or not. If not, will check the callback of EfiCreateEventReadyToBootEx.

Reply:
Got it. You are right, EHL UEFI Payload needs to be updated too to register the callback. Will submit a patch in UEFI Payload project.

Reply:
A PR for UEFI Payload project was just submitted.